src/Compilers/Core/Portable/SourceGeneration/Nodes/TransformNode.cs

```diff
@@ -79,7 +79,6 @@ public NodeStateTable<TOutput> UpdateStateTable(DriverStateTable.Builder builder
                 }
             }
 
-            Debug.Assert(newTable.Count == totalEntryItemCount);
```

Comment:
this assert was just not safe.  in a transform we can be transforming 1:N (where N is 0, 1 or >2).  So we can't make a determination about how many items teh final count is.  It's till good to prime the number of items based on what we previously saw, for the 1:1 case though.